nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -183,15 +180,15 @@ def _iter_regions_from_spheres(seeds, target_affine, target_shape, radius,
     seeds: List of triplets of coordinates in native space
         Seed definitions. List of coordinates of the seeds in the same space
         as target_affine (either of mask or if mask is None MNI space).
-    target_affine: Affine in which the regions/spheres should be projected to
-    target_shape: The shape of the output image.
     radius: float
         Indicates, in millimeters, the radius for the sphere around the seed.
     allow_overlap: boolean
         If False, an error is raised if the maps overlaps (ie at least two
         maps have a non-zero value for the same voxel).
+    mask_img: Niimg-like object,
+        Mask as a reference space to project the spheres back into brain space.
     """
-    adjacency = _get_spheres_rows(seeds, target_affine, target_shape, radius,
+    adjacency = _get_spheres_rows(seeds, mask_img.affine, mask_img.shape, radius,
                           allow_overlap)
 
     for row in adjacency.rows:
```

Comment:
I'm not sure if there are cases where this Error will be raised, and I am not sure how big of an issue this is. 

The inverse tranform might create sphere's which are possibly completely masked in the last-step. Without any warning what-so-ever, whereas an Error will occur in the classic transform.

I am not sure (yet) how to implement masking during _get_spheres_rows, but I can give it a try - If it's necessary